cmake/onnxruntime_python.cmake

```diff
@@ -370,6 +369,7 @@ add_custom_command(
 )
 
 if (NOT onnxruntime_MINIMAL_BUILD AND NOT onnxruntime_EXTENDED_MINIMAL_BUILD 
+                                  AND NOT onnxruntime_ENABLE_TRAINING
```

Comment:
why is this if block needed?
seems like onnxruntime_providers_shared is copied for the various EP cases line 430-468 below?

Reply:
This block was added for Cheng's text_execution_provider (a custom external provider) to ensure that 'onnxruntime_providers_shared' existed in his case. With training enabled, his test_execution_provider fails to build the provider bridge (as the bridge does depend on training), so the simple fix was to just disable it when training was enabled. I talked to Cheng about it and he recommended this as the fix (plus disabling the python test when training is enabled).

Reply:
So, when training is enable, how will the CUDA EP work if it can't find the bridge shared library?